Packs/DarktraceASM/Integrations/DarktraceASMRisk/DarktraceASMRisk.py

```diff
@@ -0,0 +1,1011 @@
+        return 4
+    return 1
+
+
+"""*****COMMAND FUNCTIONS****"""
+
+
+def test_module(client: Client, first_fetch_time: float) -> str:
+    """
+    Returning 'ok' indicates that the integration works like it is supposed to. Connection to the service is successful.
+
+    :type client: ``Client``
+    :param client:
+        Darktrace Client.
+    :type first_fetch_time: ``float``
+    :param first_fetch_time:
+        First fetch time.
+    :return:
+        A message to indicate the integration works as it is supposed to.
+    :rtype: ``str``
+    """
+    try:
+        first_fetch_datetime = datetime.fromtimestamp(first_fetch_time)
+        client.get_asm_risks(start_time=first_fetch_datetime)
+
+    except DemistoException as e:
+        if 'Forbidden' in str(e):
+            return 'Authorization Error: make sure API Key is correctly set'
+        else:
+            raise e
+    return 'ok'
+
+
+def fetch_incidents(client: Client,
+                    last_run: dict[str, str],
+                    first_fetch_time: float,
+                    max_alerts: int,
+                    min_severity: int,
+                    alert_types: list[str]) -> tuple[dict[str, Any], list[dict]]:
+    """Function used to pull incidents into XSOAR every few minutes.  """
+    # Get the last fetch time, if exists
+    # last_run is a dict with a single key, called last_fetch
+    last_fetch = last_run.get('last_fetch', None)
+    first_fetch_datetime = datetime.fromtimestamp(first_fetch_time)
+    # Handle first fetch time
+    if last_fetch is None:
+        last_fetch_datetime = first_fetch_datetime
+    else:
+        last_fetch_datetime = datetime.strptime(last_fetch, DATE_FORMAT)
+
+    latest_created_time = last_fetch_datetime
+
+    # Each incident is a dict with a string as a key
+    incidents: List[Dict[str, Any]] = []
+
+    asm_risks: List[Dict[str, Any]] = client.get_asm_risks(start_time=last_fetch_datetime)
+
+    for alert in asm_risks:
+        # Convert startedAt time to datetime object and add to alert
+        incident_created_time = datetime.strptime(alert['node']['startedAt'][:19], "%Y-%m-%dT%H:%M:%S")
+
+        # to prevent duplicates, we are only adding incidents with creation_time > last fetched incident
+        if last_fetch_datetime and incident_created_time <= last_fetch_datetime:
+            continue
+
+        brand = alert.get('node', {}).get('asset', {}).get('brand')
+        title = alert['node']['title']
+        incident_name = f'Darktrace ASM | Risk Title: {title} | Brand: {brand}'
+
+        xsoar_severity = _compute_xsoar_severity(alert['node']['securityRating'])
+
+        # Skip incidents with a lower severity score than the desired minimum
+        if xsoar_severity < min_severity:
+            continue
+
+        incident_type = alert['node']['type'].lower()
+
+        # Skip incidents with a type not included in the chosen alert types to ingest
+        if incident_type not in alert_types:
+            continue
```

Comment:
Also here, it is always a good idea to add debug messages stating why we are skipping something. This can help with debugging issues if they arise.

Reply:
adding this